### PR TITLE
Fix 519

### LINKED ISSUE #519: Release command takes forever
### Steps to reproduce

`mix release` under dev or prod

I was at 2.0.5 and after updating to 2.0.9 this happens.

### Description of issue

- What are the expected results?
Shouldn't take too long
- What version of Distillery?
2.0.9
- What OS, Erlang/Elixir versions are you seeing this issue on?
Debian and MacOS

### PR DESCRIPTION
**Pull request by Schultzer**
### Summary of changes

Prevent unnecessary recursive loop.

When moving the `:digraph.vertex/2` Check to `add_app/4` we can prevent an unnecessary recursive loop there happens in umbrella projects where we go through the same dependencies even when we already have seen them.

this fixes #519.

### Checklist

- [ ] New functions have typespecs, changed functions were updated
- [ ] Same for documentation, including moduledocs
- [ ] Tests were added or updated to cover changes
- [ ] Commits were squashed into a single coherent commit

### Licensing/Copyright

**By submitting this PR, you agree to the following statement, please read before submission!**

I certify that I own, and have sufficient rights to contribute, all source code and
related material intended to be compiled or integrated with the source code for Distillery
(the "Contribution"). My Contribution is licensed under the MIT License.

NOTE: If you submit a PR and remove the statement above, your PR will be rejected. For your PR to be
considered, it must contain your agreement to license under the MIT license.

### CODE REVIEW

File: lib/mix/lib/releases/models/app.ex [hunk 1]
```diff
@@ -22,14 +22,7 @@ defmodule Mix.Releases.App do
   @valid_start_types [:permanent, :temporary, :transient, :load, :none]
 
   @doc """
-  Create a new Application struct from an application name
-  """
-  @spec new(atom) :: nil | __MODULE__.t() | {:error, String.t()}
```
Comment (bitwalker) [inline lib/mix/lib/releases/models/app.ex]: Can you add this function back? It's part of the public API and isn't relevant to the fix for the issue this PR targets

**Comment by bitwalker** [comment]
Overall this looks like it should be the right fix, other than the one review comment I've left. I'll merge this once you've updated the PR. Thanks for looking into the issue, and for the contribution!

**Comment by Schultzer** [comment]
Done

**Comment by bitwalker** [comment]
Thanks!

**Pull request merged**